### PR TITLE
Editor transition upgrade

### LINKED ISSUE #35: Fix return value of dialogs & sheets
Due to how `Navigator.pop` works, currently our modal pickers (sheets, editor dialogs) do not differentiate between cancelling the modal UI and deleting the target object, as both result in `null` return value. This potentially might be resolved by #22 or by #37.

### LINKED ISSUE #36: Inconsistent modal sheet heights
Certain modal sheets _(story preview at home, and maybe some others)_ expand to a point above the app bar, which is undersirable. This inconsistency might be related to mixed usage of `CustomScrollView` and `SingleChildScrollView`.

### LINKED ISSUE #37: Replace editor dialogs with sheets
We widely use custom dialog popups for editing various objects. Some of the dialogs are quite complex and contain more than a few editable fields. And the small dialog by design might not be the best choice for display & manipulation of such types of UI. Perhaps custom modal sheets would fit better for editor spacing & flow.

### PR DESCRIPTION
**Pull request by alkaitagi**
* Replaced editor dialog with new editor sheet (closes #35, closes #36, closes #37).
* Cleaned up & added captions to sheet editors.
* Implemented cell-controlled transitions.

Crowdsroucing screen was removed (wasn't functional anyways) to be replaced by *translation* editor mode (see #41). Work with translation assets is messy for now (will be addressed by #42).

### New universal cell-based transition activation logic

Now all transitions have `condition` which controls activation. The condition is a simple object that stores check value, takes cell, applies comparison operator between its check value and the cell's content. If the comparison yields true, the transition gets selected.

User-selected and random-determined transitions were as well converted to work with cell-based conditions under the hood. When transition is selected by random or user's tap, its index is assigned to auto-resetting special `node`-cell. Then the cell is checked by pre-defined transition's condition. This might look like needless complication, but it enables universal logic for transition. And I've done work on editor to hide the complexity until user really needs it.

Due to some serialization problems the older story doesn't work. I've create *"Cell transition test"* story instead to demonstrate the new logic. The main benefit of cell-based logic is that decisions made at one point in the story can affect the narrative at some later stages. The cells act as story's memory, enabling rich branching narratives with converging and diverging storylines.

### CODE REVIEW

File: lib/modules/editor/widgets/transition_editor.dart [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                    value: ComparisionMode.equal,
+                    groupValue: celled.comparision,
+                    title: const Text('[=] Equal'),
+                    subtitle: const Text('Are the values exactly the same?'),
+                    onChanged: setMode,
+                  ),
+                  RadioListTile<ComparisionMode?>(
+                    value: ComparisionMode.lesser,
+                    groupValue: celled.comparision,
+                    title: const Text('[<] Lesser'),
```
Comment (aabounegm) [inline lib/modules/editor/widgets/transition_editor.dart:135]: "Less" is already a comparative adjective. "Lesser" has a different tone that is mostly used condescendingly. (don't forget to fix the enum as well)

File: lib/modules/play/screens/play.dart [hunk 2]
```diff
@@ -94,12 +95,34 @@ class PlayScreenState extends State<PlayScreen> {
...
+  Transition? makeTransition(Node node) {
+    final transitions = node.transitions;
     if (node.transitionInputSource == TransitionInputSource.random) {
       final index = Random().nextInt(transitions.length);
-      next = nodes[transitions[index].targetNodeId];
-    } else {}
-    if (next != null) scheduleAvdancement(next);
+      return transitions[index];
+    } else {
+      for (final transition in transitions.map((e) => e as CelledTransition)) {
```
Comment (aabounegm) [inline lib/modules/play/screens/play.dart:108]: How about 
```suggestion
      for (final transition in transitions.cast<CelledTransition>()) {
```

**Review by aabounegm — COMMENTED** [review]
This is probably more relevant to the original PR, but since I checked this one first (since it used the other branch as the base), I'll just write it:

First, I like that the editor sheets are now being used consistently everywhere instead of dialogs, and that the node editor is a dedicated full page.
As for the minus side, I find this "Cell" thing difficult to understand and unintuitive (starting with the name). It's important to keep in mind that most of our users are not programmers familiar with CS concepts, and additional power/customizability/features is not always a good thing. Having just one "store" for the game score is probably what the vast majority needs. Generally speaking, I'd also appreciate it if we discuss such changes (especially with Nursultan) before implementing them so that no effort goes to waste.
As for the play screen, I think the UI was somewhat better before. "Pause" doesn't make sense after the story is finished, and the score should be printed below the _dialogue_ instead of through a _dialog_ (see what I did there 😉?). There isn't really a timer displayed to the user to be "paused", and having the **Exit** button inside Pause is rather unintuitive. Lastly, the choice of an extended FAB with a speed dial instead of (part of) the app bar also feels somewhat twisted.
I apologize if my wording could be better, I just wanted to submit review and go sleep 😅

File: lib/modules/editor/utils/editor_sheet.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  required BuildContext context,
+  required String title,
+  T? initial,
+  required ValueGetter<T> onSave,
+  VoidCallback? onDelete,
+  required List<Widget> Function(
+    BuildContext context,
+    void Function(void Function()),
+  )
+      builder,
```
[the remark this answers is not in the document]
Comment (alkaitagi) [reply lib/modules/editor/utils/editor_sheet.dart:52]: Here I'm afraid utilizing the built-in type would result in big amount of boilerplate code due to the need of connection with scroll controller.

[the remark this answers is not in the document]
Comment (aabounegm) [reply lib/modules/editor/utils/editor_sheet.dart:52]: Well, we could still define our alias, but I agree it's not very valuable here

**Review by aabounegm — APPROVED** [review]

**Pull request merged**